### DISCUSSION TITLE
how to mix colors by taking adjacent colors from the color being mixed?

### DISCUSSION THREAD

**Discussion by NotMephisto** [InioX/matugen#129] [Q&A]
I want to implement a gradient overflow that takes `primary` color by picking up the closest colors from it and taking the invented color from `primary`, which `invert` does, however it makes it a dark color (although I think this can be fixed with `set_lighness`). There is no detailed information about this in the wiki. I would just like to make a nice shimmery button :c

**Comment by enessmr** [comment]
matugen is a material you color generation tool, overcustomizing it will make the button NOT look like m3. m3 is simple colors, like MS's fluent UI but with pastel colors
